File: slice-avatars.js
#!/usr/bin/env node

/**
 * Avatar Image Slicer
 * 
 * This script takes the provided avatar sprite image (6 avatars in a 2x3 grid)
 * and slices it into 6 individual avatar files.
 * 
 * Usage:
 *   1. Place the sprite image at: src/assets/avatars/sprite.png
 *   2. Run: node slice-avatars.js
 *   3. Output files: avatar-1.png through avatar-6.png
 */

const fs = require('fs');
const path = require('path');

// Check if 'sharp' is installed; if not, guide user
try {
    require.resolve('sharp');
} catch (e) {
    console.log('⚠️  sharp package not found. Installing...');
    require('child_process').execSync('npm install sharp', { cwd: __dirname, stdio: 'inherit' });
}

const sharp = require('sharp');

const SPRITE_PATH = path.join(__dirname, 'src/assets/avatars/sprite.png');
const OUTPUT_DIR = path.join(__dirname, 'src/assets/avatars');

async function sliceAvatars() {
    try {
        // Check if sprite file exists
        if (!fs.existsSync(SPRITE_PATH)) {
            console.error(`❌ Sprite file not found at ${SPRITE_PATH}`);
            console.log('\n📝 Instructions:');
            console.log('1. Copy your avatar sprite image to: src/assets/avatars/sprite.png');
            console.log('2. Ensure the image shows 6 avatars in a 2×3 grid (2 columns, 3 rows)');
            console.log('3. Run: node slice-avatars.js');
            process.exit(1);
        }

        // Get image metadata to determine dimensions
        const metadata = await sharp(SPRITE_PATH).metadata();
        const { width, height } = metadata;

        console.log(`📸 Sprite dimensions: ${width}×${height}px`);

        // Calculate slice dimensions (2 columns, 3 rows = 6 avatars)
        const sliceWidth = Math.floor(width / 2);
        const sliceHeight = Math.floor(height / 3);

        console.log(`✂️  Slicing into 6 avatars: ${sliceWidth}×${sliceHeight}px each\n`);

        // Extract each avatar
        const positions = [
            { row: 0, col: 0, id: 1 }, // top-left
            { row: 0, col: 1, id: 2 }, // top-right
            { row: 1, col: 0, id: 3 }, // middle-left
            { row: 1, col: 1, id: 4 }, // middle-right
            { row: 2, col: 0, id: 5 }, // bottom-left
            { row: 2, col: 1, id: 6 }  // bottom-right
        ];

        for (const { row, col, id } of positions) {
            const left = col * sliceWidth;
            const top = row * sliceHeight;
            const outputFile = path.join(OUTPUT_DIR, `avatar-${id}.png`);

            await sharp(SPRITE_PATH)
                .extract({ left, top, width: sliceWidth, height: sliceHeight })
                .toFile(outputFile);

            console.log(`✅ Created: avatar-${id}.png`);
        }

        console.log('\n🎉 All avatars extracted successfully!');
    } catch (error) {
        console.error('❌ Error slicing avatars:', error.message);
        process.exit(1);
    }
}

sliceAvatars();

File: slice_avatars.py
#!/usr/bin/env python3

"""
Avatar Image Slicer (Cross-platform)

This script slices the avatar sprite image (6 avatars in a 2x3 grid)
into 6 individual avatar files.

Usage:
    1. Place sprite.png at: src/assets/avatars/sprite.png
    2. Install Pillow: pip install pillow
    3. Run: python slice_avatars.py
"""

import sys
from pathlib import Path

try:
    from PIL import Image
except ImportError:
    print("❌ Pillow not found. Install with: pip install pillow")
    sys.exit(1)

SPRITE_PATH = Path('src/assets/avatars/sprite.png')
OUTPUT_DIR = Path('src/assets/avatars')

if not SPRITE_PATH.exists():
    print(f"❌ Sprite file not found at {SPRITE_PATH}")
    print("\n📝 Instructions:")
    print("1. Copy your avatar sprite image to: src/assets/avatars/sprite.png")
    print("2. Ensure the image shows 6 avatars in a 2×3 grid (2 columns, 3 rows)")
    print("3. Run: python slice_avatars.py")
    sys.exit(1)

try:
    # Load the sprite image
    sprite = Image.open(SPRITE_PATH)
    width, height = sprite.size
    print(f"📸 Sprite dimensions: {width}×{height}px")

    # Calculate slice dimensions (2 columns, 3 rows)
    slice_width = width // 2
    slice_height = height // 3
    print(f"✂️  Slicing into 6 avatars: {slice_width}×{slice_height}px each\n")

    # Extract each avatar
    positions = [
        (0, 0, 1),   # top-left
        (1, 0, 2),   # top-right
        (0, 1, 3),   # middle-left
        (1, 1, 4),   # middle-right
        (0, 2, 5),   # bottom-left
        (1, 2, 6)    # bottom-right
    ]

    for col, row, avatar_id in positions:
        left = col * slice_width
        top = row * slice_height
        right = left + slice_width
        bottom = top + slice_height

        # Crop the slice
        avatar_img = sprite.crop((left, top, right, bottom))

        # Save it
        output_file = OUTPUT_DIR / f'avatar-{avatar_id}.png'
        avatar_img.save(output_file)
        print(f"✅ Created: avatar-{avatar_id}.png")

    print("\n🎉 All avatars extracted successfully!")

except Exception as e:
    print(f"❌ Error: {e}")
    sys.exit(1)
